mcodingbot/plugins/highlights.py

```diff
@@ -2,21 +2,45 @@
 
 import asyncio
 from collections import defaultdict
-from typing import NamedTuple
+from typing import TYPE_CHECKING, NamedTuple, cast
 
 import crescent
 import hikari
 from asyncpg import UniqueViolationError
 from floodgate import FixedMapping
+from toolbox.members import calculate_permissions
 
 from mcodingbot.config import CONFIG
 from mcodingbot.database.models import Highlight, User, UserHighlight
 from mcodingbot.utils import Context, Plugin
 
+if TYPE_CHECKING:
+    from mcodingbot.bot import Bot
+
 MAX_HIGHLIGHTS = 25
 MAX_HIGHLIGHT_LENGTH = 32
 
 
+async def has_permission(
+    bot: Bot, guild_id: int, user_id: int, channel_id: int
```

Comment:
No reason to pass bot in here. Can just do `plugin.bot` or `plugin.app` don't remember